velox/expression/tests/SignatureTemplateBinder.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Random.h>
+
+#include "velox/expression/FunctionSignature.h"
+#include "velox/expression/SignatureBinder.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::test {
+
+class SignatureTemplateBinder {
+ public:
+  SignatureTemplateBinder(
```

Comment:
I wonder if it makes sense to rename this class to ReverseSignatureBinder and have the constructor take signature and return type. Then, add `bool tryBind()` method to process the return type and fill in as many type variables as possible and return 'true' if return type was bound successfully and false otherwise. A `bindings()` will then return partially populated bindings with some entries may be null. The caller can fill in missing values with random types or anything else and use SignatureBinder::tryResolveType to get the argument types.

Reply:
Thank you for the suggestion! Updated the code according to our offline discussion.